src/languages/mathematica.js

```diff
@@ -6,61 +6,118 @@ Website: https://www.wolfram.com/mathematica/
-    relevance: 10
+  const symbol_re = /[a-zA-Z$][a-zA-Z0-9$]*/;
+  const SYMBOLS = {
+    className: 'symbol',
+    begin: /[a-zA-Z$]/,
+    keywords: {
+      $pattern: symbol_re,
+      keyword: Mathematica.SYSTEM_SYMBOLS.join(" ")
+    },
+    end: /[a-zA-Z0-9$]*/
```

Comment:
Why wouldn't this all be a single match with `begin`?  I see that the first character can be different but traditionally we'd love this with a single match regex in `begin` to make it clear we're matching a single term and that this isn't a block match with nested rules, etc.

Reply:
I will give a more elaborate explanation in a separate comment. The summary is that I constantly ran into two issues:

- When I define keywords like all languages do at the top-level and all other symbols (that have the same regex) in the `contains:` section, I didn't get highlighting for the keywords because the "other symbols matching" took precedence.
- I ran into "0 match" exceptions more than I'd like to admit

Reply:
> When I define keywords like all languages do at the top-level and all other symbols (that have the same regex) in the contains: section, I didn't get highlighting for the keywords because the "other symbols matching" took precedence.

I think a specific example would help... but each level has it's own keywords so once you drop into a contains you no longer have keywords and if you ALSO wanted the keywords to kick in you'd need to re-include them at that level.

It's possible you're inventing to new useful pattern, but I've have to see your way, the conventional way, and then the exact problem  you're trying to solve to comment further - so a more detailed example would be appreciated. :)

Reply:
> I ran into "0 match" exceptions more than I'd like to admit

Just avoid any regex that can make a 0 width match.  :-)  That means complex regex with 20 optional parts (maybe digit) need to be pinned down to ONE core match... (like ALWAYS matching \d somewhere, etc)... often this can result in variants.  ie`\d(\.\d)?` and `.\d` vs `[+-]?(\d)?(\.\d)`.  This problem originally dates back to a complex digit match with all optional regex groups.